docs/snippets/react/my-component-play-function-alt-queries.js.mdx

```diff
@@ -0,0 +1,31 @@
+```js
+// MyComponent.stories.js | MyComponent.stories.jsx | MyComponent.stories.ts | MyComponent.stories.tsx
+
+// These are placeholders until the addon-interaction is out
+import userEvent from '@testing-library/user-event';
+import { screen } from '@testing-library/react';
+
+import { MyComponent } from './MyComponent.js';
+
+export default {
+  component: MyComponent,
+  // 👇 Marking the onButtonClick with this configuration will log the event in the Actions panel
+  argTypes: {
+    onButtonClick: { action: true },
+  },
+};
+
+export const ExampleWithRole = {
+  play: async () => {
+    // The play function interacts with the component and looks for the button.
+    await userEvent.click(screen.getByRole('button', { name: / button label/i }));
```

Comment:
This is a much better way of querying for elements than `getByTestId`. 👍

Reply:
Agreed :)